tokio/src/runtime/time/mod.rs

```diff
@@ -281,19 +295,17 @@ impl Handle {
 
                     waker_list.wake_all();
 
-                    lock = self.inner.lock();
+                    lock = self.inner.lock_sharded_wheel(id);
                 }
             }
         }
-
-        lock.next_wake = lock
-            .wheel
+        let next_wake_up = lock
             .poll_at()
             .map(|t| NonZeroU64::new(t).unwrap_or_else(|| NonZeroU64::new(1).unwrap()));
-
         drop(lock);
 
         waker_list.wake_all();
+        next_wake_up
```

Comment:
Now I'm confused that waker_list.wake_all is called potentially multiple times before self.inner.set_next_wake is called and not afterwards, but I will admit I haven't quite caught up my understanding of the implications in the multi threaded case.